apache/incubator-zeppelin
# ZeppelinHub notebook storage/connection repository

### What is this PR for?
This is to add [ZeppelinHub](https://www.zeppelinhub.com) notebook storage/connection layer to the Zeppelin. 


### What type of PR is it?
Feature

### Todos
* [x] - NotebookRepo rest api
* [x] - ZeppelinHub websocket client
* [x] - Zeppelin websocket client
* [x] - More QA (authentication consistency, etc.)
* [x] - Address review comments


### What is the Jira issue?


### How should this be tested?
First of all, you may need to create account in [ZeppelinHub](https://www.zeppelinhub.com).
Then you can set connection by following guides in [here](https://github.com/khalidhuseynov/incubator-zeppelin/blob/feat/zeppelinhub-storage/docs/storage/storage.md#notebook-storage-in-zeppelinhub--).
Finally you should be able to access and manipulate your notebooks from inside of your ZeppelinHub account.

### Screenshots (if appropriate)

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? Yes
